src/dodal/utils.py

```diff
@@ -125,16 +126,19 @@ def make_all_devices(
     if isinstance(module, str) or module is None:
         module = import_module(module or __name__)
     factories = collect_factories(module, include_skipped)
-    devices: dict[str, AnyDevice] = invoke_factories(factories, **kwargs)
+    devices: Tuple[Dict[str, AnyDevice], Dict[str, Exception]] = invoke_factories(
+        factories, **kwargs
+    )
 
     return devices
 
 
 def invoke_factories(
     factories: Mapping[str, AnyDeviceFactory],
     **kwargs,
-) -> Dict[str, AnyDevice]:
+) -> Tuple[Dict[str, AnyDevice], Dict[str, Exception]]:
     devices: dict[str, AnyDevice] = {}
+    exception_devices: dict[str, Exception] = {}
```

Comment:
Nit: I think this makes more sense

```suggestion
    device_exceptions: dict[str, Exception] = {}
```